backend/src/models/db/index.js

```diff
@@ -39,58 +26,15 @@ class DB {
                     throw new Error('Couldn\'t connect to DB')
                 }
                 this.connections.push(connection)
-            })
-        }
-    }
-
-    async init() {
-        return new Promise((resolve, reject) => {
-            // Database initialisation benchmarking
-            const timeList = [Date.now() / 1000]
-
-            // Do the initialisation
-            const [connection] = this.connections
-
-            connection.on('connect', (err) => {
-                if (err) reject(err)
 
-                // Create the database and initialise data with no dependencies.
-                const request = new Request(sqlTables(), async (err) => {
-                    if (err) reject(err)
-
-                    // Initialise with UNSW data
-                    if (!await unswDataInitialised(connection)) {
-                        await sqlUniversity(connection)
-                        await sqlFaculties(connection)
-                        await sqlDegrees(connection)
-                        await sqlSubjects(connection)
-                        await sqlCourses(connection)
-                        await sqlSessions(connection)
-                    }
-
-                    // user and question tables get dropped on init, so we must always re-add them
-                    await sqlAdminUsers(connection)
-                    await sqlQuestions(connection)
-
-                    // Initialise testing data
-                    // uses review table to check if initialization has happened
-                    if (!PRODUCTION && !await reviewTestDataInitialised(connection)) {
-                        await sqlReviews(connection)
-                        await sqlComments(connection)
-                        await sqlLikes(connection)
-                        await sqlUsers(connection)
-                    }
-
-                    // Log completion time
-                    timeList.push(Date.now() / 1000)
-                    console.log(`Done creating database! (${((timeList[1] - timeList[0])).toFixed(3)})`)
-
-                    // Done
-                    resolve()
-                })
-                connection.execSql(request)
+                // emit ready event once only when a connection becomes available
+                if (!this.ready) {
```

Comment:
does `this.ready` need to exist? Assume it's used elsewhere

Reply:
Can't see where this used unless just to be stateful

Reply:
it's used to ensure the ready event is only fired once (because a bunch of connections actually get created)
it's not used elsewhere